### PR TITLE
Provides tracing ability for local development environment

### PR DESCRIPTION
**Pull request by akeem**
# Description

* Add default instrumentation 
* Add jaeger to development docker compose file to get things started

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by akeem** [comment]
This is to get the ball rolling, instrumentation the rest of the application will be done a piece meal effort as instrumenting the application will take a while

File: locksmith/src/app.ts [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
 import express from 'express'
+import tracing from '@opencensus/nodejs'
+import { JaegerTraceExporter } from '@opencensus/exporter-jaeger'
+
+const env = process.env.NODE_ENV || 'development'
+const config = require('../config/config')[env]
+
+const exporter = new JaegerTraceExporter(config.jaeger)
+
+if (env != 'test') {
```
Comment (akeem) [inline locksmith/src/app.ts]: There maybe a slightly better place for this, but reason for being here remains the same. The tracing library intercepts the HTTP library in order to provide details on requests, timings, etc. 

Our test using the testing library `supertest` in order to provide a reasonable way  to assert on request being made during the test. 

To be expected, these together create unexpected results

File: docker/docker-compose.development.yml [hunk 2]
```diff
@@ -6,4 +6,15 @@ services:
        context: ./development
        dockerfile: ganache.dockerfile
      ports:
-       - 8545:8545
\ No newline at end of file
+       - 8545:8545
+   jager:
```
Comment (cnasc) [inline docker/docker-compose.development.yml]:
```suggestion
   jaeger:
```

File: locksmith/src/app.ts (hunk 1 above)
Comment (cnasc) [reply locksmith/src/app.ts]: I guess there's no real downside to having this not run in the test environment, right?

Comment (akeem) [reply locksmith/src/app.ts]: I don' think so, but it was a bit weird and unexpected to see this behavior

**Pull request merged**